checklists/checklistmap.php

```diff
@@ -16,109 +17,148 @@
-				var vIcon = new google.maps.MarkerImage("../images/google/smpin_red.png");
-				var pIcon = new google.maps.MarkerImage("../images/google/smpin_blue.png");
-				<?php
-				$mCnt = 0;
-				foreach($coordArr as $tid => $cArr){
-					foreach($cArr as $pArr){
-						?>
-						var pt = new google.maps.LatLng(<?php echo $pArr['ll']; ?>);
-						llBounds.extend(pt);
-						<?php
-						if(array_key_exists('occid',$pArr)){
-							?>
-							var m<?php echo $mCnt; ?> = new google.maps.Marker({position: pt, map:map, title:"<?php echo $pArr['notes']; ?>", icon:vIcon});
-							google.maps.event.addListener(m<?php echo $mCnt; ?>, "click", function(){ openIndPU(<?php echo $pArr['occid']; ?>); });
-							<?php
-						}
-						else{
-							?>
-							var m<?php echo $mCnt; ?> = new google.maps.Marker({position: pt, map:map, title:"<?php echo $pArr['sciname']; ?>", icon:pIcon});
-							<?php
-						}
-						$mCnt++;
-					}
-				}
-			}
-			$clMeta = $clManager->getClMetaData();
-			if($clMeta['footprintwkt']){
-				//Add checklist polygon
-				$wkt = $clMeta['footprintwkt'];
-				if(substr($wkt,0,7) == 'POLYGON') $wkt = substr($wkt,7);
-				else if(substr($wkt,0,12) == 'MULTIPOLYGON') $wkt = substr($wkt,12);
-				$coordArr = explode('),(', $wkt);
-				foreach($coordArr as $k => $polyFrag){
-					if($pointArr = explode(',', trim($polyFrag,' (),'))){
-						echo 'var polyPointArr'.$k.' = [];';
-						foreach($pointArr as $pointStr){
-							$llArr = explode(' ', trim($pointStr));
-							if($llArr[0] > 90 || $llArr[0] < -90) break;
-							?>
-							var polyPt = new google.maps.LatLng(<?php echo $llArr[0].','.$llArr[1]; ?>);
-							polyPointArr<?php echo $k; ?>.push(polyPt);
-							llBounds.extend(polyPt);
-							<?php
-						}
-						?>
-						var footPoly<?php echo $k; ?> = new google.maps.Polygon({
-							paths: polyPointArr<?php echo $k; ?>,
-							strokeWeight: 2,
-							fillOpacity: 0.4,
-							map: map
-						});
-						<?php
-					}
-				}
-			}
-			?>
-			map.fitBounds(llBounds);
-			map.panToBounds(llBounds);
-		}
+	include_once($SERVER_ROOT.'/includes/leafletMap.php');
+	include_once($SERVER_ROOT.'/includes/googleMap.php');
+?>
 
-		function openIndPU(occId){
-			if(puWin != null) puWin.close();
-			var puWin = window.open('../collections/individual/index.php?occid='+occId,'indspec' + occId,'scrollbars=1,toolbar=0,resizable=1,width=900,height=600,left=20,top=20');
-			if(puWin.opener == null) puWin.opener = self;
-			setTimeout(function () { puWin.focus(); }, 0.5);
-			return false;
-		}
+<script type="text/javascript" src="<?php echo $CLIENT_ROOT; ?>/js/jquery.js"></script>
+<script src="<?php echo $CLIENT_ROOT; ?>/js/symb/wktpolygontools.js" type="text/javascript"></script>
+<script type="text/javascript"> 
 
-	</script>
-	<style>
-		html, body, #map_canvas {
-			width: 100%;
-			height: 100%;
-			margin: 0;
-			padding: 0;
-		}
-	</style>
+   let voucherCoords;
```

Comment:
Little torn about this cause it will likely make the code harder to maintain. Also this js is only supposed to be use in this file and exporting it makes it re usable which means it could be misused in the future. @egbot might have a good opinion about this